backup.sh

```diff
@@ -280,39 +309,45 @@ then
 elif [ $selected_action = 'Restore' ]
 then
   if [ ! -v archive_path ]; then
-    text_input "Please provide the location of the backup archive to restore (drag-n-drop):" archive_path
+    text_input "Please provide the location of the backup directory/archive to restore (drag-n-drop):" archive_path
   fi
 
-  if [ ! -f "$archive_path" ]; then
+  if [ ! -f "$archive_path" -a "${archive_path##*.}" = "7z" ]; then
       cecho "The specified backup location doesn't exist or isn't a file."
       exit 1
+  # If we're restoring from uncompressed backup, make sure at lease the Apps directory exists
```

Comment:
```suggestion
  # If we're restoring from uncompressed backup, make sure at least the Apps directory exists
```

kinda nitpicking here